src/scheduler/mod.rs

```diff
@@ -13,20 +13,16 @@
 
 use alloc::vec_deque::VecDeque;
 
-use sync::IrqLock;
 use interrupts::{Context, SLEEP_INT};
+use smp::current;
 
 use self::thread::{KThread, State, TICKS};
 
 mod thread;
 
-// the scheduler must be created at runtime, but will not be called until
-// interrupts are enabled, which must be after the scheduler is created.
-/// The sole `Scheduler` instance
-static SCHEDULER: IrqLock<Option<Scheduler>> = IrqLock::new(None);
-
 /// Basic round-robin scheduler
-struct Scheduler {
+// pub so it can be seen by CpuLocal
```

Comment:
Is this really necessary?

Reply:
It's either that or put the scheduler definition in the smp module. I think this way makes more sense but I'm open to suggestions

Reply:
So, sorry about the confusion. I was talking about the comment, it seems redundant. The scheduler being public makes perfect sense

Reply:
Gotcha, will remove.